src/devices/cpu/capricorn/capricorn.cpp

```diff
@@ -0,0 +1,1517 @@
+// license:BSD-3-Clause
+// copyright-holders:F. Ulivi
+//
+// *****************************
+// Emulator for HP Capricorn CPU
+// *****************************
+//
+#include "emu.h"
+#include "capricorn.h"
+#include "debugger.h"
+
+// Register indexes
+// GP registers are named "R" & the octal representation of the index (00-77)
+enum {
+	CAPRICORN_R00,
+	CAPRICORN_R01,
+	CAPRICORN_R02,
+	CAPRICORN_R03,
+	CAPRICORN_R04,
+	CAPRICORN_R05,
+	CAPRICORN_R06,
+	CAPRICORN_R07,
+	CAPRICORN_R10,
+	CAPRICORN_R11,
+	CAPRICORN_R12,
+	CAPRICORN_R13,
+	CAPRICORN_R14,
+	CAPRICORN_R15,
+	CAPRICORN_R16,
+	CAPRICORN_R17,
+	CAPRICORN_R20,
+	CAPRICORN_R21,
+	CAPRICORN_R22,
+	CAPRICORN_R23,
+	CAPRICORN_R24,
+	CAPRICORN_R25,
+	CAPRICORN_R26,
+	CAPRICORN_R27,
+	CAPRICORN_R30,
+	CAPRICORN_R31,
+	CAPRICORN_R32,
+	CAPRICORN_R33,
+	CAPRICORN_R34,
+	CAPRICORN_R35,
+	CAPRICORN_R36,
+	CAPRICORN_R37,
+	CAPRICORN_R40,
+	CAPRICORN_R41,
+	CAPRICORN_R42,
+	CAPRICORN_R43,
+	CAPRICORN_R44,
+	CAPRICORN_R45,
+	CAPRICORN_R46,
+	CAPRICORN_R47,
+	CAPRICORN_R50,
+	CAPRICORN_R51,
+	CAPRICORN_R52,
+	CAPRICORN_R53,
+	CAPRICORN_R54,
+	CAPRICORN_R55,
+	CAPRICORN_R56,
+	CAPRICORN_R57,
+	CAPRICORN_R60,
+	CAPRICORN_R61,
+	CAPRICORN_R62,
+	CAPRICORN_R63,
+	CAPRICORN_R64,
+	CAPRICORN_R65,
+	CAPRICORN_R66,
+	CAPRICORN_R67,
+	CAPRICORN_R70,
+	CAPRICORN_R71,
+	CAPRICORN_R72,
+	CAPRICORN_R73,
+	CAPRICORN_R74,
+	CAPRICORN_R75,
+	CAPRICORN_R76,
+	CAPRICORN_R77,
+	CAPRICORN_ARP,
+	CAPRICORN_DRP,
+	CAPRICORN_E
+};
+
+// Bit manipulation
+static constexpr unsigned BIT_MASK(unsigned n)
+{
+	return 1U << n;
+}
+
+template<typename T> void BIT_CLR(T& w , unsigned n)
```

Comment:
Please place these utilities in an anonymous namespace to prevent symbol export.  Out-of-line instantiations of templates with the same names in different files cause very hard-to-debug issues.